variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/factories/VariantAggregatedVcfFactory.java

```diff
@@ -404,5 +406,38 @@ protected void addGenotypeWithGTS(Variant variant, VariantSourceEntry sourceEntr
             }
         }
     }
+
+    @Override
+    protected void checkVariantInformation(Variant variant, String fileId, String studyId)
+            throws NonVariantException, IncompleteInformationException {
+        VariantSourceEntry variantSourceEntry = variant.getSourceEntry(fileId, studyId);
+        if (!isVariant(variantSourceEntry)) {
+            throw new NonVariantException("The variant " + variant + " has allele frequency or counts '0'");
+        } else if (!canAlleleFrequenciesBeCalculated(variantSourceEntry)) {
+            throw new IncompleteInformationException(variant);
+        }
+    }
+
+    @Override
+    protected boolean isVariant(VariantSourceEntry variantSourceEntry) {
+        return !isAttributeZeroInVariantSourceEntry(variantSourceEntry, "AF") &&
+               !isAttributeZeroInVariantSourceEntry(variantSourceEntry, "AC") &&
+               !isAttributeZeroInVariantSourceEntry(variantSourceEntry, "AN");
```

Comment:
I can't stop thinking about this function. My spider-senses is telling me it's wrong, but I'm not sure why.

If the fields are not present, `isAttributeZeroInVariantSourceEntry` would return false, right? and then `!false && !false && !false` is true, so isVariant returns true. is that what we want?

Reply:
maybe in the case of aggregated, it's redundant with `canAlleleFrequenciesBeCalculated`? or should be done there?

Reply:
This is intended. If we have a VCF line with all the tags correctly populated, but it says: AF=0 or AC=0 (for one of the alleles if is multi-allelic), then that line does not contain a variant and the NonVariantException is thrown. If those three fields are not present, then it may be a variant or not, we cannot say "is not a variant", but we can say "the information is not complete", so we throw an IncompleteInformationException.

Maybe checkVariantInformation makes more sense if I change the order:

```
if (!canAlleleFrequenciesBeCalculated(variantSourceEntry)) {
            throw new IncompleteInformationException(variant);
} else if (!isVariant(variantSourceEntry)) {
            throw new NonVariantException("The variant " + variant + " has allele frequency or counts '0'");
       } 
```

Reply:
well, I would prefer what cyenyxe said, just put both checks as independent `if`s